### PR TITLE
Justice proxy access

### PR DESCRIPTION
**Pull request by will-freshworks**
Making the site work from https://dev.justice.gov.bc.ca/interpreter-scheduling for https://justice.gov.bc.ca/jira/browse/CSBISS-45

### CODE REVIEW

File: .docker/mock-justice-proxy/httpd.conf [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+# Proxy for Court Interpreter Scheduling project CSBISS-45
+SSLProxyEngine on
+<Location /interpreter-scheduling/api>
+ProxyPass     		 http://api:4000
+ProxyPassReverse     http://api:4000
+</Location>
+<Location /interpreter-scheduling>
+ProxyPass     		 http://client:3000
+ProxyPassReverse     http://client:3000
+</Location>
```
Comment (michel-fresh) [inline .docker/mock-justice-proxy/httpd.conf:561]: you must have spent some time typing this

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
 
   # Build the API
   build:
-    name: Build API and APP
+    name: Build API and APP and proxied APP
```
Comment (michel-fresh) [inline .github/workflows/deploy.yml:28]: 💥

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -57,6 +57,10 @@ jobs:
         working-directory: "./devops"
         run: |
           make oc-build-app
+      - name: Build proxied APP/Front-end Images
+        working-directory: "./devops"
```
Comment (michel-fresh) [inline .github/workflows/deploy.yml:61]: oh ok so this is a devops

File: Makefile [hunk 4]
```diff
@@ -100,6 +100,7 @@ e2e-test:
 
 run-dev:
 	@echo "+\n++ Make: Running dev ...\n+"
+	@docker-compose -f docker-compose.dev.yml build
```
Comment (michel-fresh) [inline Makefile:103]: `up --build` not a thing?

File: docker-compose.dev.yml [hunk 5]
```diff
@@ -106,18 +107,27 @@ services:
...
-    depends_on:
-      - client
-    environment:
-      - CYPRESS_baseUrl=http://client:3000
-    working_dir: /e2e
-    volumes:
-      - ./:/e2e
-    networks:
-      - backend
+  # cypress:
```
Comment (michel-fresh) [inline docker-compose.dev.yml:110]: could omit? kind of redundant here as we have the .test compose file

File: docker-compose.dev.yml [hunk 6]
```diff
@@ -88,8 +88,9 @@ services:
       context: ./frontend/
       dockerfile: Dockerfile.dev
     environment:
+      PUBLIC_URL: http://localhost:5000/interpreter-scheduling
       REACT_APP_REALM_AUTH_URL: http://keycloak.local.freshworks.club:8080/auth
-      REACT_APP_API_HOST: http://keycloak.local.freshworks.club:4000/api/v1
+      REACT_APP_API_HOST: http://localhost:5000/interpreter-scheduling/api/v1
```
Comment (michel-fresh) [inline docker-compose.dev.yml:93]: not .local?

File: api/src/location/location.module.ts [hunk 7]
```diff
@@ -1,15 +1,28 @@
...
+      if (!process.env.MAP_BOX_TOKEN) {
+        return {
+          provide: LocationFetchScheduleService,
+          useFactory: () => undefined
+        }
+      } else {
+        return LocationFetchScheduleService
+      }
+    })()
+  ]
```
Comment (michel-fresh) [inline api/src/location/location.module.ts:26]: curious what was going on here

**Review by michel-fresh — APPROVED** [review]
not a very meaningful approval as the oc commands are not familiar to me but the env vars for the client look good 🦄

**Review by jerryfreshworks — APPROVED** [review]
not sure using the proxy is because of avoiding CORS?

I remember create-react-app has a setting for proxy. but not sure if it applies to this situation.
https://create-react-app.dev/docs/proxying-api-requests-in-development/

File: api/package.json [hunk 8]
```diff
@@ -10,8 +10,8 @@
     "build": "nest build",
     "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
     "start": "nest start",
-    "start:dev": "nest start --watch",
-    "start:debug": "nest start --debug --watch",
+    "start:dev": "nest start --watch --preserveWatchOutput",
+    "start:debug": "nest start --debug --watch --preserveWatchOutput",
```
Comment (jerryfreshworks) [inline api/package.json:14]: cool! 💯  didn't notice this option `preserveWatchOputput`  before. can avoid clearing the terminal on change.

File: api/src/location/location.module.ts (hunk 7 above)
Comment (jerryfreshworks) [reply api/src/location/location.module.ts:26]: I check the nest document, 
maybe we could just use 
```
{
 provide: LocationFetchScheduleService,
 useFactory: !process.env.MAP_BOX_TOKEN ? undefined : LocationFetchScheduleService
 }
```

Comment (jerryfreshworks) [reply api/src/location/location.module.ts:26]: @michel-fresh my understanding is to check if `MAP_BOX_TOKEN` exists, if so, then include the service, 
if not, then do nothing.

Correct me if I'm wrong.

File: Makefile (hunk 4 above)
Comment (will-freshworks) [reply Makefile:103]: slipped my mind, will change

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (will-freshworks) [reply .github/workflows/deploy.yml:61]: this is how you tell yes

File: docker-compose.dev.yml (hunk 6 above)
Comment (will-freshworks) [reply docker-compose.dev.yml:93]: Only matters for the keycloak URL

File: api/src/location/location.module.ts (hunk 7 above)
Comment (will-freshworks) [reply api/src/location/location.module.ts:26]: I tried those approaches above @jerryfreshworks and they each failed for different reasons. Weirdly, the approach I've used seemed to be the simplest. Though, please check it out for yourself if you are curious. Possibly I am not enough of a nest guru to get it working this way.

Comment (will-freshworks) [reply api/src/location/location.module.ts:26]: And yeah, to answer @michel-fresh 's question, Jerry is right. It's a conditionally injected service. Seemed like the simplest approach.

**Comment by will-freshworks** [comment]
@jerryfreshworks 
> I remember create-react-app has a setting for proxy. but not sure if it applies to this situation.

No this is slightly different. I can explain in stand-up tomorrow if you'd like.

File: api/package.json (hunk 8 above)
Comment (fwpushan) [reply api/package.json:14]: 👍

**Pull request merged**